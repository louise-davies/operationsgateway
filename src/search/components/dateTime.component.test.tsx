import React from 'react';
import DateTime, {
  DateTimeSearchProps,
  datesEqual,
  verifyAndUpdateDate,
  type VerifyAndUpdateDateParams,
} from './dateTime.component';
import { render, RenderResult, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import {
  applyDatePickerWorkaround,
  cleanupDatePickerWorkaround,
} from '../../setupTests';

describe('datesEqual function', () => {
  it('returns true if both dates are null', () => {
    const date1 = null;
    const date2 = null;
    expect(datesEqual(date1, date2)).toBe(true);
  });

  it('returns true if both dates are Invalid Date', () => {
    const date1 = new Date('');
    const date2 = new Date('');
    expect(datesEqual(date1, date2)).toBe(true);
  });

  it('returns true if one date is invalid and the other is null', () => {
    const date1 = null;
    const date2 = new Date('');
    expect(datesEqual(date1, date2)).toBe(true);
  });

  it('returns true if dates are the same date', () => {
    const date1 = new Date('2019-09-18');
    const date2 = new Date('2019-09-18');
    expect(datesEqual(date1, date2)).toBe(true);
  });

  it('returns false if dates are not the same date', () => {
    const date1 = new Date('2019-09-18');
    const date2 = new Date('2019-09-19');
    expect(datesEqual(date1, date2)).toBe(false);
  });

  it('returns false if one date is invalid and the other is valid', () => {
    const date1 = new Date('2019-09-18');
    const date2 = new Date('');
    expect(datesEqual(date1, date2)).toBe(false);
  });

  it('returns false if one date is null and the other is valid', () => {
    const date1 = null;
    const date2 = new Date('2019-09-18');
    expect(datesEqual(date1, date2)).toBe(false);
  });
});

describe('verifyAndUpdateDate function', () => {
  let props: VerifyAndUpdateDateParams;
  const changeDate = jest.fn();

  beforeEach(() => {
    props = {
      date: new Date('2022-01-01T00:00:00'),
      prevDate: null,
      otherDate: null,
      fromDateOrToDateChanged: 'fromDate',
      changeDate: changeDate,
    };
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('calls changeDate if otherDate is null', () => {
    verifyAndUpdateDate(props);

    expect(changeDate).toHaveBeenCalledWith(new Date('2022-01-01 00:00:00'));
  });

  it('calls changeDate if valid fromDate', () => {
    verifyAndUpdateDate({
      ...props,
      otherDate: new Date('2022-01-02T00:00:00'),
    });

    expect(changeDate).toHaveBeenCalledWith(new Date('2022-01-01 00:00:00'));
  });

  it('calls changeDate if valid toDate', () => {
    verifyAndUpdateDate({
      ...props,
      otherDate: new Date('2021-01-01T00:00:00'),
      fromDateOrToDateChanged: 'toDate',
    });

    expect(changeDate).toHaveBeenCalledWith(new Date('2022-01-01 00:00:00'));
  });

  it('allows changeDate to be called if both dates are equal', () => {
    verifyAndUpdateDate({
      ...props,
      otherDate: new Date('2022-01-01T00:00:00'),
    });

    expect(changeDate).toHaveBeenCalledWith(new Date('2022-01-01T00:00:00'));
  });

  it("doesn't call changeDate if fromDate is invalid", () => {
    verifyAndUpdateDate({
      ...props,
      otherDate: new Date('2021-01-01T00:00:00'),
    });

    expect(changeDate).not.toHaveBeenCalled();
  });

  it("doesn't call changeDate if toDate is invalid", () => {
    verifyAndUpdateDate({
      ...props,
      otherDate: new Date('2022-01-02T00:00:00'),
      fromDateOrToDateChanged: 'toDate',
    });

    expect(changeDate).not.toHaveBeenCalled();
  });

  it("doesn't call changeDate if date hasn't changed based on prevDate", () => {
    verifyAndUpdateDate({
      ...props,
      prevDate: props.date,
    });

    expect(changeDate).not.toHaveBeenCalled();
  });
});

describe('DateTime tests', () => {
  let props: DateTimeSearchProps;
  const changeSearchParameterFromDate = jest.fn();
  const changeSearchParameterToDate = jest.fn();
  const resetTimeframe = jest.fn();
  const resetExperimentTimeframe = jest.fn();
  const setExperimentTimeframe = jest.fn();

  const createView = (): RenderResult => {
    return render(<DateTime {...props} />);
  };

  beforeEach(() => {
    applyDatePickerWorkaround();
    userEvent.setup();
    props = {
      searchParameterFromDate: null,
      searchParameterToDate: null,
      changeSearchParameterFromDate,
      changeSearchParameterToDate,
      resetTimeframe,
      timeframeRange: null,
      resetExperimentTimeframe,
      searchParameterExperiment: null,
      setExperimentTimeframe,
      experiments: [],
      shotnumToDate: undefined,
    };
  });

  afterEach(() => {
    cleanupDatePickerWorkaround();
    jest.clearAllMocks();
  });

  it('renders correctly with no input date-time ranges', () => {
    const view = createView();

    expect(view.asFragment()).toMatchSnapshot();
  });

  it('renders correctly with input date-time ranges', () => {
    props.searchParameterFromDate = new Date('2021-01-01 00:00:00');
    props.searchParameterToDate = new Date('2021-01-02 00:00:00');

    const view = createView();

    expect(view.asFragment()).toMatchSnapshot();
  });

  it('calls changeDate and resetTimeframe and resetExperimentTimeframe when filling out and clearing date-time inputs', async () => {
    const experiments = [
      {
        _id: '17210011-1',
        end_date: '2019-01-20T18:00:00',
        experiment_id: '17210011',
        part: 1,
        start_date: '2019-01-07T10:00:00',
      },
      {
        _id: '18110022-1',
        end_date: '2019-05-21T16:48:00',
        experiment_id: '18110022',
        part: 1,
        start_date: '2019-04-08T09:00:00',
      },
      {
        _id: '18110023-1',
        end_date: '2019-02-12T17:48:00',
        experiment_id: '18110023',
        part: 1,
        start_date: '2019-01-21T10:00:00',
      },
    ];
    props.experiments = experiments;
    props.searchParameterExperiment = experiments[0];

    createView();

    const dateFilterFromDate = screen.getByLabelText('from, date-time input');
    await userEvent.type(dateFilterFromDate, '2022-01-01 00:00:00');
    expect(changeSearchParameterFromDate).toHaveBeenCalledWith(
      new Date('2022-01-01 00:00:00')
    );

    const dateFilterToDate = screen.getByLabelText('to, date-time input');
    await userEvent.type(dateFilterToDate, '2022-01-02 00:00:00');
    expect(changeSearchParameterToDate).toHaveBeenCalledWith(
      new Date('2022-01-02 00:00:00')
    );

    expect(resetTimeframe).toHaveBeenCalled();
    expect(resetExperimentTimeframe).toHaveBeenCalled();

    await userEvent.clear(dateFilterFromDate);
    expect(changeSearchParameterFromDate).toHaveBeenCalledWith(null);

    await userEvent.clear(dateFilterToDate);
    expect(changeSearchParameterToDate).toHaveBeenCalledWith(null);
  });

  it('calls changeDate and resetTimeframe and setExperimentTimeframe when filling out and clearing date-time inputs', async () => {
    const experiments = [
      {
        _id: '17210011-1',
        end_date: '2019-01-20T18:00:00',
        experiment_id: '17210011',
        part: 1,
        start_date: '2019-01-07T10:00:00',
      },
      {
        _id: '18110022-1',
        end_date: '2019-05-21T16:48:00',
        experiment_id: '18110022',
        part: 1,
        start_date: '2019-04-08T09:00:00',
      },
      {
        _id: '18110023-1',
        end_date: '2019-02-12T17:48:00',
        experiment_id: '18110023',
        part: 1,
        start_date: '2019-01-21T10:00:00',
      },
    ];
    props.experiments = experiments;
    props.searchParameterExperiment = experiments[0];

    createView();

    const dateFilterFromDate = screen.getByLabelText('from, date-time input');
    await userEvent.type(dateFilterFromDate, '2019-01-07 11:00:00');
    expect(changeSearchParameterFromDate).toHaveBeenCalledWith(
      new Date('2019-01-07 11:00:00')
    );

    const dateFilterToDate = screen.getByLabelText('to, date-time input');
    await userEvent.type(dateFilterToDate, '2019-01-20 17:00:00');
    expect(changeSearchParameterToDate).toHaveBeenCalledWith(
      new Date('2019-01-20 17:00:00')
    );

    expect(resetTimeframe).toHaveBeenCalled();
    expect(setExperimentTimeframe).toHaveBeenCalled();

    await userEvent.clear(dateFilterFromDate);
    expect(changeSearchParameterFromDate).toHaveBeenCalledWith(null);

    await userEvent.clear(dateFilterToDate);
    expect(changeSearchParameterToDate).toHaveBeenCalledWith(null);
  });

  it.todo('calls changeDate when opening calendar and selecting elements');

  it('displays helper text while typing date-time (to date)', async () => {
    createView();

    const dateFilterToDate = screen.getByLabelText('to, date-time input');
    await userEvent.type(dateFilterToDate, '2022-01-');
    expect(
      screen.getByText('Date-time format: yyyy-MM-dd HH:mm')
    ).toBeInTheDocument();
  });

  it('displays helper text while typing date-time (from date)', async () => {
    createView();

    const dateFilterFromDate = screen.getByLabelText('from, date-time input');
    await userEvent.type(dateFilterFromDate, '2022-01-');
    expect(
      screen.getByText('Date-time format: yyyy-MM-dd HH:mm')
    ).toBeInTheDocument();
  });

  it('handles invalid date-time values correctly by not calling changeDate and displaying helper text', async () => {
    createView();

    const dateFilterFromDate = screen.getByLabelText('from, date-time input');
    await userEvent.type(dateFilterFromDate, '2022-01-01 00:00:00');

    expect(changeSearchParameterFromDate).toHaveBeenLastCalledWith(
      new Date('2022-01-01 00:00:00')
    );

    const dateFilterToDate = screen.getByLabelText('to, date-time input');
    await userEvent.type(dateFilterToDate, '2021-01-01 00:00:00');

    const helperTexts = screen.getAllByText('Invalid date-time range');

    // One helper text below each input
    expect(helperTexts.length).toEqual(2);

    expect(changeSearchParameterToDate).not.toHaveBeenCalled();
  });
});
