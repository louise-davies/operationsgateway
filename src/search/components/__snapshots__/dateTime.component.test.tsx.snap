// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomPickersDay function renders the CustomPickerDay correctly 1`] = `
<body>
  <div>
    <div
      class="MuiPickersDay-root MuiPickersDay-dayWithMargin MuiPickersDay-hiddenDaySpacingFiller MuiPickersDay-hiddenDaySpacingFiller css-1dh00js-MuiPickersDay-root"
    />
  </div>
</body>
`;

exports[`DateTime tests renders correctly with input date-time ranges 1`] = `
<DocumentFragment>
  <div
    aria-label="date-time search box"
    class="MuiBox-root css-whpmxb"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1cprn3g-MuiSvgIcon-root"
      data-testid="CalendarMonthIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 4h-1V2h-2v2H8V2H6v2H5c-1.11 0-1.99.9-1.99 2L3 20c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2m0 16H5V10h14zM9 14H7v-2h2zm4 0h-2v-2h2zm4 0h-2v-2h2zm-8 4H7v-2h2zm4 0h-2v-2h2zm4 0h-2v-2h2z"
      />
    </svg>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1vpwcmr-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-29c4sz-MuiTypography-root"
        >
          From date
        </p>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-n5zogd-MuiFormControl-root-MuiTextField-root"
        >
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-953pxc-MuiInputBase-root-MuiInput-root"
          >
            <input
              aria-invalid="false"
              aria-label="from, date-time input"
              autocomplete="off"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd css-jflll1-MuiInputBase-input-MuiInput-input"
              id="from date-time"
              inputmode="text"
              placeholder="From..."
              type="text"
              value="2021-01-01 00:00"
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1673wpy-MuiInputAdornment-root"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall clearButton css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                title="Clear value"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                  data-testid="ClearIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
            >
              <button
                aria-label="from, date-time picker"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeSmall css-1bkxqye-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="CalendarIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth MuiDivider-vertical MuiDivider-flexItem css-fx8517-MuiDivider-root"
    />
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1vpwcmr-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-29c4sz-MuiTypography-root"
        >
          To date
        </p>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-n5zogd-MuiFormControl-root-MuiTextField-root"
        >
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-953pxc-MuiInputBase-root-MuiInput-root"
          >
            <input
              aria-invalid="false"
              aria-label="to, date-time input"
              autocomplete="off"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd css-jflll1-MuiInputBase-input-MuiInput-input"
              id="to date-time"
              inputmode="text"
              placeholder="To..."
              type="text"
              value="2021-01-02 00:00"
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1673wpy-MuiInputAdornment-root"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall clearButton css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                title="Clear value"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                  data-testid="ClearIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
            >
              <button
                aria-label="to, date-time picker"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeSmall css-1bkxqye-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="CalendarIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`DateTime tests renders correctly with no input date-time ranges 1`] = `
<DocumentFragment>
  <div
    aria-label="date-time search box"
    class="MuiBox-root css-whpmxb"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1cprn3g-MuiSvgIcon-root"
      data-testid="CalendarMonthIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 4h-1V2h-2v2H8V2H6v2H5c-1.11 0-1.99.9-1.99 2L3 20c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2m0 16H5V10h14zM9 14H7v-2h2zm4 0h-2v-2h2zm4 0h-2v-2h2zm-8 4H7v-2h2zm4 0h-2v-2h2zm4 0h-2v-2h2z"
      />
    </svg>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1vpwcmr-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-29c4sz-MuiTypography-root"
        >
          From date
        </p>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-n5zogd-MuiFormControl-root-MuiTextField-root"
        >
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-953pxc-MuiInputBase-root-MuiInput-root"
          >
            <input
              aria-invalid="false"
              aria-label="from, date-time input"
              autocomplete="off"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd css-jflll1-MuiInputBase-input-MuiInput-input"
              id="from date-time"
              inputmode="text"
              placeholder="From..."
              type="text"
              value=""
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
            >
              <button
                aria-label="from, date-time picker"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeSmall css-1bkxqye-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="CalendarIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth MuiDivider-vertical MuiDivider-flexItem css-fx8517-MuiDivider-root"
    />
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1vpwcmr-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-29c4sz-MuiTypography-root"
        >
          To date
        </p>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-n5zogd-MuiFormControl-root-MuiTextField-root"
        >
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-953pxc-MuiInputBase-root-MuiInput-root"
          >
            <input
              aria-invalid="false"
              aria-label="to, date-time input"
              autocomplete="off"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd css-jflll1-MuiInputBase-input-MuiInput-input"
              id="to date-time"
              inputmode="text"
              placeholder="To..."
              type="text"
              value=""
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
            >
              <button
                aria-label="to, date-time picker"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeSmall css-1bkxqye-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="CalendarIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renderExperimentPickerDay function renders a PickersDay component when selectedDate is not with an experiment 1`] = `
<Memo
  autoFocus={false}
  day={2022-02-05T00:00:00.000Z}
  disabled={false}
  isAnimating={false}
  isFirstVisibleCell={false}
  isLastVisibleCell={false}
  onBlur={[MockFunction]}
  onDaySelect={[MockFunction]}
  onFocus={[MockFunction]}
  onKeyDown={[MockFunction]}
  outsideCurrentMonth={true}
  selected={false}
  today={false}
/>
`;

exports[`renderExperimentPickerDay function renders a PickersDay component when selectedDate is null 1`] = `
<Memo
  autoFocus={false}
  day={2022-02-05T00:00:00.000Z}
  disabled={false}
  isAnimating={false}
  isFirstVisibleCell={false}
  isLastVisibleCell={false}
  onBlur={[MockFunction]}
  onDaySelect={[MockFunction]}
  onFocus={[MockFunction]}
  onKeyDown={[MockFunction]}
  outsideCurrentMonth={true}
  selected={false}
  today={false}
/>
`;

exports[`renderExperimentPickerDay function renders a PickersDay component when selectedDate is with an experiment 1`] = `
<Styled(memo)
  autoFocus={false}
  day={2022-02-05T00:00:00.000Z}
  dayIsBetween={false}
  disableMargin={true}
  disabled={false}
  isAnimating={false}
  isFirstDay={false}
  isFirstVisibleCell={false}
  isLastDay={false}
  isLastVisibleCell={false}
  onBlur={[MockFunction]}
  onDaySelect={[MockFunction]}
  onFocus={[MockFunction]}
  onKeyDown={[MockFunction]}
  outsideCurrentMonth={true}
  selected={false}
  today={false}
/>
`;
